xbmc/filesystem/ZipFile.cpp

```diff
@@ -403,8 +403,29 @@ int CZipFile::UnpackFromMemory(std::string& strDest, const std::string& strInput
         return iResult;
       if( (mZipItem.flags & 8) == 8 )
       {
-        CLog::Log(LOGERROR,"FileZip: extended local header, not supported!");
-        return iResult;
+        // if an extended local header is present, the following fields are 0
+        // in the local header and we need to read them from the extended local header
+
+        // search for the extended local header
+        unsigned int i = iPos + LHDR_SIZE + mZipItem.flength + mZipItem.elength;
+        while (1) 
+        {
+          if (i + 16 > strInput.size()) 
+          {
+            CLog::Log(LOGERROR,"FileZip: extended local header expected, but not present!");
+            return iResult;
+          }
+          if ((strInput[i] == 0x50) && (strInput[i + 1] == 0x4b) && 
+            (strInput[i + 2] == 0x07) && (strInput[i + 3] == 0x08))
+            break;  // header found
+          i++;
+        }
+        mZipItem.crc32 = strInput[i + 4] || strInput[i + 5] << 8 || strInput[ i+ 6] << 16 || 
```

Comment:
Shouldn 't this be "|" to compute the checksum?

Reply:
Yes... I know, embarrassing. Found it myself and fixed it in the second commit.
Should I have squashed my commits? (sorry, it's my first PR)

Reply:
Yes, please. Sorry - I read from top to bottom.

Reply:
Will do in my next PR :thumbsup:

Be careful with the left-shifts - at least on my platform (Visual Studio) these were _signed char_. As a result, any bytes bigger than 0x80 caused problems: `(signed char)0x81 << 8 == 0xffff8100`. That's why I had add the cast to _(uint8_t)_